src/main/java/com/example/job_service/config/SecurityConfig.java

```diff
@@ -0,0 +1,62 @@
+package com.example.job_service.config;
+
+import com.example.job_service.service.CustomUserDetailService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+
+
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig {
+
+    @Autowired
+    CustomUserDetailService customUserDetailService;
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+
+        return http.csrf(AbstractHttpConfigurer::disable)
+                .authorizeHttpRequests(auth -> auth
+                        .requestMatchers("/", "/register","/login").permitAll()
+                        .requestMatchers("/admin").hasRole("USER")
+                        .anyRequest()
+                        .authenticated()
+                )
+                .formLogin(login -> login
+                        .loginPage("/login").permitAll()
+                                .failureUrl("/login?error=ture")
```

Comment:
spelling mistake

Reply:
sorry, which spelling mistake?

Reply:
Shouldn’t it be error =true?

Reply:
oh no no it’s called ternary operator i don’t exactly know if i have used correctly or not but i think it basically means if error = true then redirect to login page, (? mean if the ?’s right side is true then do the ?’s left side part)

Reply:
Yes that I understood already. I wasn’t sure if “ture” is a keyword.